sway_libs/src/fixed_point/ufp/SPECIFICATION.md

```diff
@@ -0,0 +1,86 @@
+# Overview
+
+This document provides an overview of the Unsigned Fixed Point library.
+
+It outlines the use cases, i.e. specification, and describes how to implement the library.
+
+## Use Cases
+
+The Unsigned Fixed Point type can be used anytime a developer needs fixed-point numbers.
+
+## Public Functions
+
+### `bits()`
+
+The size of this type in bits.
+
+### `max()`
+
+The largest value that can be represented by this fixed-point type.
+
+### `min()`
+
+The smallest value that can be represented by this fixed-point type.
+
+### `round()`
+
+Returns the nearest integer to this fixed-point type. 
+
+### `recip()`
+
+Takes the reciprocal (inverse) of a number, `1/x`.
+
+### `trunc()`
+
+Returns the integer part of a fixed-point number. 
+
+### `from()`
+
+Creates a fixed-point value from the underlying type, as is.
+
+### `into()`
+
+Returns a the underlying value of a fixed-point number, as is.
+
+### `denominator()`
+
+Returns a the value of the standard denominator of a fixed-point number, in the underlying type.
+
+### `zero()`
+
+Returns a zero value of type.
+
+### `from_uint()`
+
+Returns a fixed-point value equal to some unsigned value.
+
+### `floor()`
+
+Returns the biggest integer fixed-point value, that is smaller than the argument.

```

Comment:
Is this biggest or smallest? Same for `ceil()`

Reply:
Biggest integer that is smaller than the number i.e. 100.5 has 101 integer smaller than it, but `floor` has to return 100

Reply:
Nvm I think this is correct